taf/api/targets.py

```diff
@@ -662,3 +662,83 @@ def _update_target_repos(
         path = targets_dir / target_repo_name
         path.write_text(json.dumps(data, indent=4))
         taf_logger.info(f"Updated {path}")
+
+
+@check_if_clean
+def taf_status(path: str, library_dir: Optional[str] = None) -> None:
+    """
+    Prints a list of target repositories of an authentication repository, their states,
+    and the dependencies of the authentication repository.
+
+    Arguments:
+        path: Authentication repository's location
+        library_dir (optional): Path to the library's root directory. Determined based on the authentication repository's path if not provided.
+
+    Side Effects:
+       None
+
+    Returns:
+        None
+    """
+    # Get the authentication repository status
+    auth_repo = AuthenticationRepository(path=path)
+    head_commit = auth_repo.head_commit_sha()
+    if head_commit is None:
+        print("Repository is empty")
+        return
+    top_commit = [head_commit]
+
+    # Load repositories and target repositories data
+    repositoriesdb.load_repositories(auth_repo)
+    target_repositories = repositoriesdb.get_deduplicated_repositories(auth_repo)
+    repositories_data = auth_repo.sorted_commits_and_branches_per_repositories(
+        top_commit
+    )
+
+    # Print authentication repository status
+    print(f"Authentication Repository: {path}")
```

Comment:
Here you can't use this path, because it might be `.`, but you can print the repository's name: `auth_repo.name` or full path `auth_repo.path` (here we resolve the path, so we know we won't print the relative path).